pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -44,14 +50,46 @@
  */
 public class PinotCatalog implements Schema {
 
+  public static final String DEFAULT_DB_NAME = "default";
+
   private final TableCache _tableCache;
 
+  private final Map<String, CalciteSchema> _subCatalog;
+
+  private final String _databaseName;
+
   /**
    * PinotCatalog needs have access to the actual {@link TableCache} object because TableCache hosts the actual
    * table available for query and processes table/segment metadata updates when cluster status changes.
    */
   public PinotCatalog(TableCache tableCache) {
     _tableCache = tableCache;
+    _databaseName = null;
+    // create all databases
+    // TODO: we need to monitor table cache changes to register newly created databases
+    // TODO: we also need to monitor table that needs to be put into the right places
+    _subCatalog = constructSubCatalogs(_tableCache);
+  }
+
+  private PinotCatalog(String databaseName, TableCache tableCache) {
+    _tableCache = tableCache;
+    _databaseName = databaseName;
+    _subCatalog = null;
+  }
+
+  private Map<String, CalciteSchema> constructSubCatalogs(TableCache tableCache) {
+    Map<String, CalciteSchema> subCatalog = new HashMap<>();
+    for (String physicalTableName : tableCache.getTableNameMap().keySet()) {
+      String[] nameSplit = StringUtils.split(physicalTableName, '.');
```

Comment:
There used to be a feature flag to allow only 1 dot in table names, and if that was not enabled pinot didn’t allow dot in table names. The feature flag was recently [removed](https://github.com/apache/pinot/pull/12402) and allowing 1 dot was made the general behaviour in favour of supporting database in pinot. That said any earlier table name with a dot will implicitly fall under the database that is prefixed by the dot.

Reply:
So it is not backward compatible, right? :D 

We need to document that very well.

At the same time, as I said otherwise, in SQL schemas (which is what we are implementing here) can be recursive, so more than one `.` should be also reserved.

At the same time there is a standard SQL way to add this characters and is to use specify the table name within double quotes.

Reply:
> So it is not backward compatible, right? :D
Querying the older tables with dot won't break because of this change. Can you explain why you think its not backward compatible?

About the recursive schema, we don’t have a strong usecase for it now. Should be able to extend when there is one.

Reply:
So before the option was deprecated table names could either:
  * Do no have `.` in their name
  * Or have a single `.` in their name, in which case it was used to identify the _database_

And now, independent of that property, table names may have a single `.` in their name, in which case it was used to identify the _database_.

Then I guess it is not backward incompatible.

Reply:
Yeah that's correct 👍